code/game/gamemodes/shadowling/shadowling.dm

```diff
@@ -0,0 +1,212 @@
+	- Enemy numbers
+	- Lasers (Lasers are concentrated light and do more damage)
+	- Flashbangs (High stun and high burn damage; if the light stuns humans, you bet your ass it'll hurt the shadowling very much!)
+
+Shadowlings start off disguised as normal crew members, and they only have two abilities: Hatch and Enthrall.
+They can still enthrall and perhaps complete their objectives in this form.
+Hatch will, after a short time, cast off the human disguise and assume the shadowling's true identity.
+They will then assume the normal shadowling form and gain their abilities.
+
+The shadowling will seem OP, and that's because it kinda is. Being restricted to the dark while being alone most of the time is extremely difficult and as such the shadowling needs powerful abilities.
+Made by Xhuis
+
+*/
+
+
+
+/*
+	GAMEMODE
+*/
+
+
+/datum/game_mode
+	var/list/datum/mind/shadows = list()
+	var/list/datum/mind/thralls = list()
+	var/list/shadow_objectives = list()
+	var/required_thralls = 15 //How many thralls are needed (hardcoded for now)
+	var/shadowling_ascended = 0 //If at least one shadowling has ascended
+	var/shadowling_dead = 0 //is shadowling kill
+
+
+/datum/game_mode/shadowling
+	name = "shadowling"
+	config_tag = "shadowling"
+	antag_flag = BE_SHADOWLING
+	required_players = 0 //30
+	required_enemies = 1
+	recommended_enemies = 3
+	restricted_jobs = list("AI", "Cyborg")
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain")
+
+/datum/game_mode/shadowling/announce()
+	world << "<b>The current game mode is - Shadowling!</b>"
+	world << "<b>There are alien <span class='userdanger'>shadowlings</span> on the station. Crew: Kill the shadowlings before they can eat or enthrall the crew. Shadowlings: Enthrall the crew while remaining in hiding.</b>"
+
+/datum/game_mode/shadowling/pre_setup()
+	if(config.protect_roles_from_antagonist)
+		restricted_jobs += protected_jobs
+
+	if(config.protect_assistant_from_antagonist)
+		restricted_jobs += "Assistant"
+
+	for(var/datum/mind/player in antag_candidates)
+		for(var/job in restricted_jobs)
+			if(player.assigned_role == job)
+				antag_candidates -= player
+
+	var/shadowlings = 3 //How many shadowlings there are; hardcoded to 3 for now
+
+	var/datum/mind/shadow = pick(antag_candidates)
+	shadows += shadow
+	antag_candidates -= shadow
+	modePlayer += shadow
+	shadow.special_role = "Shadowling"
+	shadowlings--
+	return 1
+
+
+/datum/game_mode/shadowling/post_setup()
+	for(var/datum/mind/shadow in shadows)
+		log_game("[shadow.key] (ckey) has been selected as a Shadowling.")
+		sleep(10)
+		shadow.current << "<span class='deadsay'><b>You are a shadowling!</b></span>"
+		greet_shadow(shadow)
+		finalize_shadowling(shadow)
+		process_shadow_objectives(shadow)
+		//give_shadowling_abilities(shadow)
+	..()
+	return
+
+/datum/game_mode/proc/greet_shadow(var/datum/mind/shadow)
+	shadow.current << "<b>Currently, you are disguised as an employee aboard Space Station 13.</b>"
```

Comment:
can't you just use [world.name]?

On Fri, Mar 6, 2015 at 8:23 PM, Cheridan <notifications@github.com> wrote:

> In code/game/gamemodes/shadowling/shadowling.dm
> <https://github.com/tgstation/-tg-station/pull/8157#discussion_r25970025>:
>
> > +
> > +
> > +/datum/game_mode/shadowling/post_setup()
> > +	for(var/datum/mind/shadow in shadows)
> > +		log_game("[shadow.key] (ckey) has been selected as a Shadowling.")
> > +		sleep(10)
> > +		shadow.current << "<span class='deadsay'><b>You are a shadowling!</b></span>"
> > +		greet_shadow(shadow)
> > +		finalize_shadowling(shadow)
> > +		process_shadow_objectives(shadow)
> > +		//give_shadowling_abilities(shadow)
> > +	..()
> > +	return
> > +
> > +/datum/game_mode/proc/greet_shadow(var/datum/mind/shadow)
> > +	shadow.current << "<b>Currently, you are disguised as an employee aboard Space Station 13.</b>"
>
> In this case, no it's not. If the server sets the name as Space Station 13
> then it will be consistent but otherwise yeah it should use station name.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/8157/files#r25970025>.
>